### PR TITLE
Implemented coolest interface.

### PR DESCRIPTION
**Pull request by nkhadka21**
This PR implements coolest interface. One can use implemented function to update coolest template file.

### CODE REVIEW

File: tests/TestData/lens_class.pickle
Comment (sibirrer) [inline tests/TestData/lens_class.pickle:1]: do you need a 7MB test file?

File: slsim/Util/coolest_slsim_interface.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    update_coolest_from_lenstronomy,
+    create_lenstronomy_from_coolest,
+)
+from slsim.Util.param_util import magnitude_to_amplitude, amplitude_to_magnitude
+
+
+def update_coolest_from_lenstronomy_slsim(
+    lens_class, path, file_name, band=None, mag_zero_point=27, ending="_update"
+):
+    """This function updates given coolest format .json file using lenstronomy kwargs of
```
Comment (sibirrer) [inline slsim/Util/coolest_slsim_interface.py:11]: why do you need a lenstronomy in the function name and description? At the end you have a slsim_lens object and what you get is a coolest file. Under the hook as it is coded right now, it uses lenstronomy, but that is not important for the user using this functionality

File: slsim/Util/coolest_slsim_interface.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    )
+    return update_coolest
+
+
+def create_lenstronomy_slsim_from_coolest(path, file_name, mag_zero_point=27):
+    """This function creates an lenstronomy_kwargs from the coolest file.
+
+    :param path: path to the .json file that need to be updated
+    :param file_name: name of the .json file without .json extension
+    :returns: lenstronomy_kwargs for slsim
```
Comment (sibirrer) [inline slsim/Util/coolest_slsim_interface.py:72]: the return is not specific enough documented

File: tests/test_coolest_slsim_interface.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+from slsim.Util.coolest_slsim_interface import (
+    update_coolest_from_lenstronomy_slsim,
+    create_lenstronomy_slsim_from_coolest,
+)
+import pickle
+import os
+
+
+def test_update_coolest_from_lenstronomy_slsim():
```
Comment (sibirrer) [inline tests/test_coolest_slsim_interface.py:9]: can you make a test going both ways first generating a coolest file, then reading it into a slsim object and then back, and compare whether the back and forth results in the same outcome?

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you very much @nkhadka21. I added some feedback with a request of an additional test

File: slsim/Util/coolest_slsim_interface.py (hunk 1 above)
Comment (nkhadka21) [reply slsim/Util/coolest_slsim_interface.py:11]: Changed!

**Comment by nkhadka21** [comment]
Hi @sibirrer , I worked on your feedback. Now it is ready for review!

File: slsim/lens.py [hunk 4]
```diff
@@ -361,7 +361,7 @@ def los_linear_distortions(self):
             gamma2 = gamma * np.sin(2 * phi)
             self._gamma = [gamma1, gamma2]
         if not hasattr(self, "_kappa"):
-            self._kappa = np.random.normal(loc=0, scale=0.05)
+            self._kappa = abs(np.random.normal(loc=0, scale=0.05))
```
Comment (sibirrer) [inline slsim/lens.py:364]: kappa really has to go also negative. The average convergence must be =0 in the universe

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you, Narayan! It looks good other than the convergence values. They have to be allowed to get negative. We might just not test the feature yet for. negative values and raise and issue in COOLEST, but it should not be our problem to deal with this.

File: notebooks/coolest_template_generator.ipynb [hunk 5]
```diff
@@ -0,0 +1,132 @@
+{
```
Comment (sibirrer) [inline notebooks/coolest_template_generator.ipynb:1]: <p>do we need this notebook? It basically demonstrates COOLEST parts, and not SLSim aspects. The current notebook has not a single SLSim line of code used.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/LSST-strong-lensing/slsim/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (nkhadka21) [reply notebooks/coolest_template_generator.ipynb]: Deleted for now. I can add it later with slsim example (updating coolest and creating slsim).

File: slsim/lens.py (hunk 4 above)
Comment (nkhadka21) [reply slsim/lens.py]: Avoided test for now.

**Review by sibirrer — APPROVED** [review]
Thank you @nkhadka21

**Pull request merged**